src/object/srv_csum.c

```diff
@@ -507,7 +507,7 @@ ds_csum_add2iod(daos_iod_t *iod, struct daos_csummer *csummer,
 		ci_insert(&iod_csums->ic_data[0], 0,
 			   biov_csums[0].cs_csum, biov_csums[0].cs_len);
 		if (biov_csums_used != NULL)
-			(*biov_csums_used)++;
+			(*biov_csums_used) = 1;
```

Comment:
seems some problems for the cases that multiple IODs and some are array and some are singv,
csum_add2iods -> ds_csum_add2iod(), here will reset biov_csums_used as I once there is one singv IOD in the middle?

Reply:
biov_csums_used should be reset each time ds_csum_add2iod() is entered. The issue this change resolves is singe value wasn't resetting it each time.